File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    w = 3
    h = (brown + yellow) // w
    area = brown + yellow
    while (w * h) != (brown + yellow) or yellow != (w-2)*(h-2):
        w += 1
        h = (brown + yellow) // w

    answer = [max(w, h), min(w, h)]
    return answer

File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
       int[] answer = new int[2];

        int sum = brown + yellow;

        for (int i = 3; i < sum; i++) {
            int h = sum / i;
            if (((h - 2) * (i - 2) == yellow)) {
                answer[0] = i;
                answer[1] = h;
            }
        }

        return answer;
    }
}